src/features/events/components/LocationModal/LocationDetailsCard.tsx

```diff
@@ -1,23 +1,59 @@
 import { Close } from '@mui/icons-material';
-import { FC } from 'react';
-import { Box, Button, Typography } from '@mui/material';
+import {
+  Box,
+  Button,
+  ClickAwayListener,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { FC, useCallback, useEffect, useState } from 'react';
 
+import LocationsModel from 'features/events/models/LocationsModel';
 import messageIds from 'features/events/l10n/messageIds';
 import { useMessages } from 'core/i18n';
 import { ZetkinLocation } from 'utils/types/zetkin';
+import ZUIPreviewableInput, {
+  ZUIPreviewableMode,
+} from 'zui/ZUIPreviewableInput';
 
 interface LocationDetailsCardProps {
+  model: LocationsModel;
   onClose: () => void;
   onUseLocation: () => void;
   location: ZetkinLocation;
 }
 
 const LocationDetailsCard: FC<LocationDetailsCardProps> = ({
+  model,
   onClose,
   onUseLocation,
   location,
 }) => {
   const messages = useMessages(messageIds);
+  const [title, setTitle] = useState(location.title);
+  const [description, setDescription] = useState(location.info_text);
+  const [fieldEditing, setFieldEditing] = useState<
+    'title' | 'description' | null
+  >(null);
+
+  const handleDescriptionTextAreaRef = useCallback(
```

Comment:
Why is `ZUIPreviewableInput` not already handling this?

Reply:
For some reason I do not comprehend, it set focus at the start of the input when it is in multiline mode, so i just stole this from your work on the notes in views.